.github/workflows/run-pytest.yml

```diff
@@ -30,22 +30,23 @@ jobs:
           - os: 'macos-latest'
             python: '3.12'
     steps:
-    - uses: actions/checkout@v4
+    - name: Checkout
+      uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
       with:
         submodules: "recursive"
     - name: Set up Python ${{ matrix.python }}
-      uses: actions/setup-python@v5
+      uses: actions/setup-python@39cd14951b08e74b54015e9e001cdefcf80e669f # v5.1.1
```

Comment:
isn't it better to have a specific version? can dependabot update the setup if you select a specific hash instead of a version? Is there a reason why this exact one?

Reply:
Josh is a proponent of specific hashes (for reproducibility as far as I know) but I agree it gives up some clarity